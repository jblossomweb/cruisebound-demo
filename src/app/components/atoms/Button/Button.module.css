.btn {
  @apply font-normal rounded;
}

.btn-shadow {
  @apply drop-shadow-lg;
}

.btn-tiny {
  @apply text-xs;
  padding: 2px 4px;
}

.btn-small {
  @apply text-sm py-1 px-2;
}

.btn-medium {
  @apply text-base py-2 px-3;
}

.btn-large {
  @apply py-3 px-4
    text-sm
    sm:text-base
    md:text-lg
    lg:text-2xl;
}

.btn-blue {
  @apply bg-blue-500 text-white;
}

.btn-blue:hover {
  @apply bg-blue-700;
}

.btn-red {
  @apply bg-red-500 text-white;
}

.btn-red:hover {
  @apply bg-red-700;
}

.btn-green {
  @apply bg-green-500 text-white;
}

.btn-green:hover {
  @apply bg-green-700;
}

.btn-slate {
  @apply bg-slate-600 text-white;
}

.btn-slate:hover {
  @apply bg-slate-700;
}

.btn-white {
  @apply bg-white
    text-black
    border-black;
  border-width: 1px;
}

.btn-white:hover {
  @apply bg-gray-100;
}
