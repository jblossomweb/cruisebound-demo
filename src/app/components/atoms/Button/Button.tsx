import React from 'react'
import classNames from 'classnames'

import styles from './Button.module.css'

export interface ButtonProps {
  text: string | React.ReactNode
  color?: 'blue' | 'red' | 'green' | 'slate' | 'white'
  size?: 'tiny' | 'small' | 'medium' | 'large'
  shadow?: boolean
  onClick?: () => void
}

const Button: React.FC<ButtonProps> = ({
  text = '',
  color = 'blue',
  size = 'medium',
  shadow = false,
  onClick = undefined,
}) => (
  <button
    type="button"
    className={classNames(
      styles.btn,
      styles[`btn-${color}`],
      styles[`btn-${size}`],
      { [styles['btn-shadow']]: shadow }
    )}
    onClick={onClick}
  >
    {text}
  </button>
)

export default Button
