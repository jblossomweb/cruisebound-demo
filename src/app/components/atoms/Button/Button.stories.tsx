import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import Button, { type ButtonProps } from './Button'

export default {
  title: 'components/atoms/Button',
  component: Button,
  argTypes: {
    text: { control: 'text' },
    color: {
      control: 'select',
      options: ['blue', 'red', 'green', 'slate', 'white'],
    },
    size: { control: 'select', options: ['tiny', 'small', 'medium', 'large'] },
    shadow: { control: 'boolean' },
  },
} as Meta<typeof Button>

const Template: StoryFn<typeof Button> = (args: ButtonProps) => (
  <Button {...args} />
)

export const Default = Template.bind({})
Default.args = {
  text: 'Default',
}

export const Blue = Template.bind({})
Blue.args = {
  text: 'Blue',
  color: 'blue',
}

export const Red = Template.bind({})
Red.args = {
  text: 'Red',
  color: 'red',
}

export const Green = Template.bind({})
Green.args = {
  text: 'Green',
  color: 'green',
}

export const Slate = Template.bind({})
Slate.args = {
  text: 'Slate',
  color: 'slate',
}

export const White = Template.bind({})
White.args = {
  text: 'White',
  color: 'white',
}

export const Tiny = Template.bind({})
Tiny.args = {
  text: 'Tiny',
  size: 'tiny',
}

export const Small = Template.bind({})
Small.args = {
  text: 'Small',
  size: 'small',
}

export const Medium = Template.bind({})
Medium.args = {
  text: 'Medium',
  size: 'medium',
}

export const Large = Template.bind({})
Large.args = {
  text: 'Large',
  size: 'large',
}

export const Shadow = Template.bind({})
Shadow.args = {
  text: 'Shadow',
  shadow: true,
}

export const NoText = Template.bind({})
NoText.args = {}
