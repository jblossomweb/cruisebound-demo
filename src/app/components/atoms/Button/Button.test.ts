import storySnaps from 'test/utils/storySnaps'
import * as stories from './Button.stories'

describe('app/components/atoms/Button', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
