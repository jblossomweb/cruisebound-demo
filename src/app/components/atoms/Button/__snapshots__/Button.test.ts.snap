// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Button stories Blue matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-medium"
    type="button"
  >
    Blue
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Default matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-medium"
    type="button"
  >
    Default
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Green matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-green btn-medium"
    type="button"
  >
    Green
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Large matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-large"
    type="button"
  >
    Large
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Medium matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-medium"
    type="button"
  >
    Medium
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories NoText matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-medium"
    type="button"
  />
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Red matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-red btn-medium"
    type="button"
  >
    Red
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Shadow matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-medium btn-shadow"
    type="button"
  >
    Shadow
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Slate matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-slate btn-medium"
    type="button"
  >
    Slate
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Small matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-small"
    type="button"
  >
    Small
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories Tiny matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-blue btn-tiny"
    type="button"
  >
    Tiny
  </button>
</DocumentFragment>
`;

exports[`app/components/atoms/Button stories White matches markup snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-white btn-medium"
    type="button"
  >
    White
  </button>
</DocumentFragment>
`;
